### PR TITLE
Do not allow node to code if any of the selected nodes are in error state

### PR DESCRIPTION
**Pull request by pinzart90**
https://jira.autodesk.com/browse/DYN-4612
Crash when one of the selected nodes is in error state.

### Release Notes

Do not allow node to code if any of the selected nodes are in error state

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs [hunk 1]
```diff
@@ -899,7 +899,12 @@ internal void NodeToCode(object parameters)
 
         internal bool CanNodeToCode(object parameters)
         {
-            return DynamoSelection.Instance.Selection.OfType<NodeModel>().Any();
+            var nodeModels = DynamoSelection.Instance.Selection.OfType<NodeModel>();
+            if (!nodeModels.Any() || nodeModels.Any(x => x.IsInErrorState))
+            {
+                return false;
+            }
+            return true;
```
[the remark this answers is not in the document]
Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs:907]: This changes the UX a bit.
image
FYI @Amoursol

**Review by aparajit-pratap — COMMENTED** [review]
@pinzart90 can we add a test to verify there is no crash if you programmatically try to convert an erroring node to code.

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs:907]: This code does not check if the Nodes are connected. So if a user has an unconnected node with an error ...and it is included in the selection..then Node To Code will still not be allowed

[the remark this answers is not in the document]
Comment (Jingyi-Wen) [reply src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs:907]: I don't think it's a big concern

File: src/DynamoUtilities/ObservableHashSet.cs [hunk 2]
```diff
@@ -24,8 +23,16 @@ public void Add(T item)
...
+            List<T> removedObjs = new List<T>();
+            foreach(var item in set.ToArray())
+            {
+                if (match(item))
+                {
+                    removedObjs.Add(item);
+                    set.Remove(item);
+                }
+            }
+            CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Remove, removedObjs));
```
Comment (pinzart90) [inline src/DynamoUtilities/ObservableHashSet.cs:35]: Looks like passing null to that specific `NotifyCollectionChangedEventArgs` constructor throws an exception.
@aparajit-pratap I tried to pass in the removed objects here. PTAL

Comment (aparajit-pratap) [reply src/DynamoUtilities/ObservableHashSet.cs:35]: Yes, I put in a fix here: https://github.com/DynamoDS/Dynamo/pull/12719

Comment (aparajit-pratap) [reply src/DynamoUtilities/ObservableHashSet.cs:35]: I think this is overkill at least now when we're not using any of this information.

**Review by aparajit-pratap — APPROVED** [review]
Other than holding off on the observablehashset change, LGTM.

**Pull request merged**